main.py

```diff
@@ -86,38 +86,41 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
              if not ok:
                 return
 
-        new_stream = {"url": stream_url, "quality": stream_quality}
-
+        streams = None
         try:
-            self.setup_videoframe(new_stream, self.coordinates)
-            self.new_coordinates()
-        except KeyError:
+            session = Streamlink()
```

Comment:
Maybe we should create the session when we're starting the application as `self.session`, and not everytime we open a new stream? To my understanding the session can be used multiple times, with different urls? 

As it is right now we shouldn't get that big of a performance gain, since we're still creating a **new** session for each added stream right?